src/strategies/optimism/stargate/strategy-stargate-base.sol

```diff
@@ -0,0 +1,196 @@
+        if (_want > 0) {
+            ILPStakingTime(stakingContract).deposit(poolId, _want);
+        }
+    }
+
+    function _withdrawSome(uint256 _amount)
+        internal
+        override
+        returns (uint256)
+    {
+        ILPStakingTime(stakingContract).withdraw(poolId, _amount);
+        return _amount;
+    }
+
+
+    // **** State Mutations ****
+    // Adds/updates a swap path from a token to native, normally used for adding/updating a reward path
+    function addToNativeRoute(ISolidlyRouter.route[] calldata path) external {
+        require(msg.sender == timelock, "!timelock");
+        _addToNativeRoute(path);
+    }
+
+    function _addToNativeRoute(ISolidlyRouter.route[] memory path) internal {
+        require(path[path.length - 1].to == native, "!valid route");
+        address token = path[0].from;
+        if (toNativeRoutes[token].length == 0) {
+            activeRewardsTokens.push(token);
+        }
+        for (uint256 i = 0; i < path.length; i++) {
+            toNativeRoutes[token].push(path[i]);
+        }
+        IERC20(token).approve(solidRouter, type(uint256).max);
+    }
+
+    function addToTokenRoute(ISolidlyRouter.route[] calldata path) external {
+        require(msg.sender == timelock, "!timelock");
+
+        _addToTokenRoute(path);
+    }
+
+    function _addToTokenRoute(ISolidlyRouter.route[] memory path) internal {
+        require(path[0].from == native, "!valid route");
+
+        address token = path[path.length - 1].to;
+        // Reset old route if necessary
+        if (toTokenRoutes[token].length > 0) {
+            delete toTokenRoutes[token];
+        }
+
+        // Set the route
+        for (uint256 i = 0; i < path.length; i++) {
+          toTokenRoutes[token].push(path[i]);
+        }
+    }
+
+    function deactivateReward(address reward) external {
+        require(msg.sender == timelock, "!timelock");
+        for (uint256 i = 0; i < activeRewardsTokens.length; i++) {
+            if (activeRewardsTokens[i] == reward) {
+                activeRewardsTokens[i] = activeRewardsTokens[activeRewardsTokens.length - 1];
+                activeRewardsTokens.pop();
+            }
+        }
+    }
+
+    function _swapSolidlyWithRoute(ISolidlyRouter.route[] memory routes, uint256 _amount) internal {
+        require(routes[0].to != address(0));
+        ISolidlyRouter(solidRouter).swapExactTokensForTokens(_amount, 0, routes, address(this), block.timestamp.add(60));
+    }
+
+    function _harvestRewards() internal {
+        ILPStakingTime(stakingContract).deposit(poolId, 0);
+    }
+
+    function _swapActiveRewardsToNative() internal {
+        for (uint256 i = 0; i < activeRewardsTokens.length; i++) {
+            uint256 _rewardToken = IERC20(activeRewardsTokens[i]).balanceOf(address(this));
+            if (toNativeRoutes[activeRewardsTokens[i]].length > 0 && _rewardToken > 0) {
+                _swapSolidlyWithRoute(toNativeRoutes[activeRewardsTokens[i]], _rewardToken);
```

Comment:
What's the motivation for using Velodrome over Uniswap, which appears to have greater liquidity on Optimism?